src/plugin.js

```diff
@@ -8,16 +8,33 @@
 
 import * as React from 'react';
 
+import {createPlugin} from 'fusion-core';
 import type {FusionPlugin, Middleware} from 'fusion-core';
 
 import Provider from './provider';
 
+// eslint-disable-next-line
+type FusionPluginNoHidden<TDeps, TService> = $Diff<
+  FusionPlugin<TDeps, TService>,
+  {__plugin__: boolean}
+>;
+
 export default {
-  create: (
+  create: <TDeps: *, TService: *>(
     name: string,
-    plugin: FusionPlugin<*, *>,
+    // eslint-disable-next-line
+    plugin: FusionPluginNoHidden<TDeps, TService> | FusionPlugin<TDeps, TService>,
```

Comment:
We allow folks to pass in non-plugins that contain `middleware`, `provides`, and `deps`.  Technically these are not `FusionPlugin`s.

Alternatively, it might be nicer to just use a custom type here:
```
type PartialPlugin<Deps, Service> = {
  deps?: Deps,
  provides?: (Deps: $ObjMap<Deps & {}, ExtractReturnType>) => Service,
  middleware?: (
    Deps: $ObjMap<Deps & {}, ExtractReturnType>,
    Service: Service
  ) => Middleware
};
...
export default { 
     ...
     plugin: PartialPlugin<TDeps, TService> | FusionPlugin<TDeps, TService>
     ...
};
```

This avoids the shenanigans of exposing `__plugin__` outside of `fusion-core`.  But also means we have lose the explicit dependency between these two types.  Thoughts?

Reply:
Where do we use non-plugins today? Should we enforce that they are Fusion plugins?

Reply:
Example usage in [src/__tests__/plugin.js](https://github.com/fusionjs/fusion-react/blob/master/src/__tests__/plugin.js#L42).  Without this, they would have to call `createPlugin` and pass in the returned value.